test/integration/clustermanager_test.go

```diff
@@ -448,20 +460,452 @@ var _ = ginkgo.Describe("ClusterManager", func() {
 							continue
 						}
 						if now.After(caCert.NotAfter) {
-							return false
+							return fmt.Errorf("certificate after NotAfter")
 						}
 						if now.Before(caCert.NotBefore) {
-							return false
+							return fmt.Errorf("certificate before NotBefore")
 						}
 						found = true
 						break
 					}
 					if !found {
+						return fmt.Errorf("not found")
+					}
+				}
+				return nil
+			}, eventuallyTimeout*3, eventuallyInterval*3).Should(gomega.BeNil())
+		})
+	})
+})
+
+var _ = ginkgo.Describe("ClusterManager Detached Mode", func() {
+	var cancel context.CancelFunc
+	var hubRegistrationDeployment = fmt.Sprintf("%s-registration-controller", clusterManagerName)
+
+	ginkgo.BeforeEach(func() {
+		var ctx context.Context
+		ctx, cancel = context.WithCancel(context.Background())
+		go startHubOperatorDetached(ctx)
+	})
+
+	ginkgo.AfterEach(func() {
+		if cancel != nil {
+			cancel()
+		}
+	})
+
+	ginkgo.Context("Deploy and clean hub component", func() {
+		ginkgo.It("should have expected resource created successfully", func() {
+			// Check namespace
+			gomega.Eventually(func() bool {
```

Comment:
return error is much better especially full debugging.

Reply:
Yes, returning error is better. Updated the newly added cases and the previous cases to give more detailed msg.

Reply:
Fixed.